pennylane/ops/op_math/product.py

```diff
@@ -0,0 +1,249 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Product class which contains logic for
+computing the product of operations.
+"""
+from functools import reduce
+
+import numpy as np
+import pennylane as qml
+
+from pennylane import math
+from pennylane.operation import Operator
+
+
+def op_prod(*Prods, do_queue=True, id=None):
+    r"""This constructs an operator that is the product of operators.
+    All operators like qml.RX, qml.RZ and qml.RY are rotation matrices, thus the product is commutative and
```

Comment:
additional to black, also run `pylint pennylane/ops/op_math/product.py` and you will see what pylint complains about. apparently there are some trailing whitespaces here :)

Reply:
Yes I tried running black but the errors did not dissapear, Im going to follow all your advice and format the code correctly. Thank you very much.